tests/fixture/test_data_server.go

```diff
@@ -139,13 +141,13 @@ func AllocateFundedKeys(baseURI string, count int) ([]*secp256k1.PrivateKey, err
 	if err != nil {
 		return nil, fmt.Errorf("failed to construct request: %w", err)
 	}
+
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
 		return nil, fmt.Errorf("failed to request funded keys: %w", err)
 	}
-	if resp.Body != nil {
-		defer resp.Body.Close()
-	}
+	defer resp.Body.Close()
```

Comment:
`Body` is guaranteed to never be `nil` (and if it wasn't, then the below `io.ReadAll(resp.Body)` would panic anyway)